lib/friends/introvert.rb

```diff
@@ -658,6 +658,19 @@ def set_n_activities!(type)
       end
     end
 
+    def set_implicit_locations(_activities)
+      implicit_location = nil
+      @activities.reverse_each do |activity|
+        # Does activity include 'moved to _location_' line?
+        if activity.moved_to_location
+          implicit_location = activity.moved_to_location
+          next
+        end
+        # Does activity have a location in description?
+        activity.implicit_location = implicit_location if activity.location_names.empty?
```

Comment:
I find it a little confusing to have `location_names` not include the implicit location. Could we rename it to something like `explicit_location_names` or `description_location_names` to make this clearer?

Reply:
Agree, I've changed to `description_location_names`

Reply:
I agree with you, this could be changed to something more explicit. I therefore tried changing all references to `location_names` to `description_location_names` - however, I got errors when running the tests, as something in the code was still somehow calling `location_names`. A little investigation with `pry` revealed this line:

**friends/lib/friends/introvert.rb:641)**
```
activity.send("#{type}_names").each do |thing_name|
``` 
We could split this one line into an `if` statement that explicity calls `description_location_names` if `type` is a `location` (else it will call `friends_names`). But I wanted to discuss with you whether that's the right thing to do first.

Reply:
Ah, nice catch. Looking at that line, it's used in `set_n_activities!` which sets the `n_activities` field so that commands like `friends list favorite locations` works correctly. Since that should include the implicit locations, I think what makes the most sense is to rename the method to `description_location_names` as you had, and make a new `location_names` method that includes both explicit and implicit location names. Does that make sense?

This is also making me realize that we should add one or two new tests to `friends list favorite locations` to confirm this behavior (and also confirm that a location can't double-count a location if it's mentioned twice either explicitly or implicitly). Does that make sense and seem right?

Reply:
Done and done - `description_location_names` has been replaced the old `location_names`, and I've created a new `location_names` with a different implementation to take into account both locations in descriptions and implicit locations.
I've also added a test for `list favorite locations`. I've also removed the location `London` from the tests I've created, because I realised they it isn't necessary